src/Runtime/Core/Collections/IntPair.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Probabilistic.Collections
+{
+    /// <summary>
+    /// Hand-rolled implementation of <see cref="ValueTuple{T1,T2}"/> for integers.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="ValueTuple{T1,T2}.Equals(ValueTuple{T1,T2})"/> uses
+    /// <see cref="EqualityComparer{T}.Default"/> to implement he comparison. This actually involves
+    /// a lookup in a hash-table each time it is used due to the way static fields in generic
+    /// classes are implemented. It appears to be measurably slower when int pair is used as
+    /// a key in a dictionary.
+    /// </remarks>
+    public struct IntPair : IEquatable<IntPair>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IntPair"/> struct.
+        /// </summary>
+        /// <param name="first">The first element of the pair.</param>
+        /// <param name="second">The second element of the pair.</param>
+        public IntPair(int first, int second)
```

Comment:
Do we really want this constructor?
A year, or so, ago I did an experiment and it turned out, initializing structs with
`new IntPair() { First = <first>, Second = <second> }`
is faster than calling a constructor that does that.

Reply:
I just checked and saw no difference. I guess it gets inlined if struct and constructor are small enough. Or my benchmarks are not good enough.

Reply:
OK, I lost the benchmark I had and I don't seem to be able to reproduce it now, so either it got fixed or was complicated enough to encounter. Anyway, not a significant problem for us now.